workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py

```diff
@@ -134,7 +145,7 @@ def rms_errors(prognostic, verification_c48, grid):
 def zonal_means_dycore(prognostic, verification, grid):
     logger.info("Preparing zonal+time means (dycore)")
     zonal_means = zonal_mean(prognostic, grid.lat)
-    return time_mean(zonal_means)
+    return _assign_source_attrs(time_mean(zonal_means), prognostic)
```

Comment:
Can you push this attribute assignment into the `time_mean` and `rms` functions? It seems the RMS function should define the fact that it doesn't change the units for example.

Reply:
Yeah I considered that. Currently the function works for either Dataset or DataArray inputs, so will need to restrict to only Datasets or do some type checking to properly assign attributes in both cases.

Reply:
or perhaps better, still use the xarray keep_attrs=True option, but just within particular contexts? I think that would be simplest.